pkg/devices/gen_net.go

```diff
@@ -59,6 +59,17 @@ func NewGenNetDevice(dev *ghw.PCIDevice, deviceID string, dt types.DeviceType, i
 			return nil, err
 		}
 		netNames, _ = utils.GetNetNames(deviceID)
+	case types.AuxNetDeviceType:
+		if pfName, err = utils.GetSriovnetProvider().GetUplinkRepresentorFromAux(deviceID); err != nil {
+			// AuxNetDeviceType by design should have PF, return error if failed to get PF name
+			return nil, err
+		}
+		if pfAddr, err = utils.GetSriovnetProvider().GetPfPciFromAux(deviceID); err != nil {
+			return nil, err
+		}
+		// Only SF auxiliary devices can have an index, for other (-1, err) returned. Ignore this error
```

Comment:
why we ignore errors?

Reply:
Because if it is not SF, then error will be returned.

Reply:
if deviceID does not point to an SF shouldnt we fail ?

what is the scenario where one or more of the methods below fail but we want NewGenNetDevice to return with no errors ?

Reply:
> if deviceID does not point to an SF shouldnt we fail ?
> 

PR is adding support for auxiliary network devices, not SFs only.

> what is the scenario where one or more of the methods below fail but we want NewGenNetDevice to return with no errors ?

GetSfIndexByAuxDev - when device is not SF (and also when `sfnum` file read/parsing failed)
GetNetDevicesFromAux - due to the same reasons as VFs, I believe

Reply:
> PR is adding support for auxiliary network devices, not SFs only.

Yes, however ATM we dont have a use-case for non SF AuxNetDevice.

moreover if the auxiliray device is a SF and we fail to get SF Index then i think we do want to fail.

Reply:
> > PR is adding support for auxiliary network devices, not SFs only.
> 
> Yes, however ATM we dont have a use-case for non SF AuxNetDevice.
> 

In such case, why had I added aux. devices support and not just SFs?)

> moreover if the auxiliray device is a SF and we fail to get SF Index then i think we do want to fail.

It will also fail if no `sfnum` file presented, because it is non-SF device. I don't want plugin to fail in such case.

But Ok, I can make it fail and put a comment to review this line in the future